### ISSUE TITLE
LÖVE API Status

### ISSUE THREAD

**Issue by bobbens** [naev/naev#1578]
To keep track of implementation status of [LÖVE API](https://love2d.org/) in Naev. Stuff that isn't here should be already implemented partially or fully.

**Want to implement:**

- [x] love.audio
- [x] love.graphics.Canvas
- [x] love.graphics.Shader
- [x] love.math.Transform (and appropriate stuff in love.graphics)

**Maybe implement:**

- [ ] love.event
- [ ] love.graphics.ParticleSystem
- [ ] love.graphics.SpriteBatch
- [ ] love.graphics.Video
- [ ] love.graphics.Text
- [ ] love.joystick
- [ ] love.sound
- [ ] love.thread
- [ ] love.video
- [x] utf8

**Not planned**

- love.graphics.Mesh
- love.font
- love.image.CompressedImageData
- love.physics
- love.math.BezierCurve
- love.touch
- love-enet
- luasocket

Currently all the demos from https://simplegametutorials.github.io/love/ work properly. You have to unzip the `.love` file and run it with something like
```
love = require 'love'
love.exec( 'path/to/gamedirectory' )
```
![out](https://user-images.githubusercontent.com/54677/102576816-e7a19600-4139-11eb-8775-27ff81938853.gif)

**Comment by UncombedCoconut** [comment]
Half-baked idea: maybe Naev's implementation should be stricter (e.g., rejecting `image:draw(..)` as a shortcut for `graphics.draw(image, ...)` or `shader:send("my_vec2", x, y)` for `shader:send("my_vec2", {x, y})`).
That way code is less likely to blow up if we want to port it to OG love2d and compare behavior.

**Comment by bobbens** [comment]
> Half-baked idea: maybe Naev's implementation should be stricter (e.g., rejecting `image:draw(..)` as a shortcut for `graphics.draw(image, ...)` or `shader:send("my_vec2", x, y)` for `shader:send("my_vec2", {x, y})`).
> That way code is less likely to blow up if we want to port it to OG love2d and compare behavior.

Mmm, while I do think we have to support Love2D API, not sure we have to limit ourselves to it. Both cases you are showing are basically me not thinking too much and implementing it. It doesn't really help that the Love2D functions are overloaded as hell and it makes all the documentation hard to follow.

**Comment by bobbens** [comment]
Rethinking it, I don't think we need events in general for Naev. For now I think the status is good enough. What I think could be interesting to implement of the "maybe implement" would be `love.sound`, and while I would like to have `love.video`, I think we to make some videos first (plus it would probably pull in ffmpeg as a dependency)

**Comment by bobbens** [comment]
Closing issue as I think nothing major is missing and we can handle small API case-by-case.
